objc/objc_runtime_darwin.go

```diff
@@ -148,203 +168,167 @@ func AllocateClassPair(super Class, name string, extraBytes uintptr) Class {
-			return 0, fmt.Errorf("objc: couldn't add Method %s", met.Name)
+		if !class.AddMethod(def.Cmd, imp, encoding) {
+			return 0, fmt.Errorf("objc: couldn't add Method at index %d", idx)
 		}
 	}
 	// Add Ivars
 	// Start at 1 because we skip the class object which is first
-	for i := 1; i < strct.NumField(); i++ {
-		f := strct.Field(i)
-		if f.Name == "_" {
-			continue
-		}
-		size := f.Type.Size()
-		alignment := uint8(math.Log2(float64(f.Type.Align())))
-		enc, err := encodeType(f.Type, false)
+	for _, ivar := range ivars {
+		size := ivar.Type.Size()
+		alignment := uint8(math.Log2(float64(ivar.Type.Align())))
+		enc, err := encodeType(ivar.Type, false)
 		if err != nil {
-			return 0, fmt.Errorf("objc: couldn't add Ivar %s: %w", f.Name, err)
+			return 0, fmt.Errorf("objc: couldn't add Ivar %s: %w", ivar.Name, err)
 		}
-		if !class_addIvar(class, f.Name, size, alignment, enc) {
-			return 0, fmt.Errorf("objc: couldn't add Ivar %s", f.Name)
+		if !class_addIvar(class, ivar.Name, size, alignment, enc) {
+			return 0, fmt.Errorf("objc: couldn't add Ivar %s", ivar.Name)
 		}
-		if offset := class.InstanceVariable(f.Name).Offset(); offset != f.Offset {
-			return 0, fmt.Errorf("objc: couldn't add Ivar %s because offset (%d != %d)", f.Name, offset, f.Offset)
+		switch ivar.Attribute {
+		case ReadWrite:
+			ty := reflect.FuncOf(
+				[]reflect.Type{
+					reflect.TypeOf(ID(0)), reflect.TypeOf(SEL(0)), ivar.Type,
+				},
+				nil, false,
+			)
+			var encoding string
+			if encoding, err = encodeFunc(reflect.New(ty).Elem().Interface()); err != nil {
+				return 0, fmt.Errorf("objc: failed to create read method for '%s': %w", ivar.Name, err)
+			}
+			instanceVariable := class.InstanceVariable(ivar.Name)
+			val := reflect.MakeFunc(ty, func(args []reflect.Value) (results []reflect.Value) {
+				// on entry the first and second arguments are ID and SEL followed by the value
+				if len(args) != 3 {
+					panic(fmt.Errorf("objc: incorrect number of arguments"))
+				}
+				// Grab the pointer from using the Ivar and dereference it
+				var value ID
+				switch args[2].Kind() {
+				case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+					value = ID(args[2].Int())
+				case reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+					value = ID(args[2].Uint())
+				case reflect.Float32, reflect.Float64:
+					value = ID(math.Float64bits(args[2].Float()))
+				case reflect.Bool:
+					if args[2].Bool() {
+						value = 1
+					}
+				case reflect.Pointer, reflect.UnsafePointer:
+					value = ID(args[2].Pointer())
+				default:
+					panic(fmt.Errorf("objc: unsupported kind %s", args[2].Kind()))
+				}
+				object_setIvar(args[0].Interface().(ID), instanceVariable, value)
+				return nil
+			}).Interface()
+			// this code only works for ascii but that shouldn't be a problem
+			if ivar.Type.Kind() == reflect.Bool && strings.HasPrefix(ivar.Name, "is") {
+				ivar.Name = ivar.Name[2:]
```

Comment:
This doesn't work when the property name is for example `island`. Maybe detecting `'A' <= ivar.Name[2] && ivar.Name[2] <= 'Z'` is enough. (I am not sure when the name is `is12345` 🤔 )

Reply:
how about only checking when the third letter is lower case then do nothing

Reply:
Maybe that works.

In the first place, should `ivar.Name` include `is`? I am not sure.

Reply:
Assuming `ivar.Name` does NOT include `is`, just adding `is` when making a getter name might make sense (though I am not sure which is more conventional). 

https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingIvarsAndTypes.html (editable vs isEditable)

Reply:
So we say that Ivar.Name should not start with is and then just append is when the type is bool?

Reply:
IMO yes. What do you think?

Reply:
I think that works